Started redesigning SI

Imipenem opened the pull request:
- BaseInfo (not quite happy with naming though): Hold Info basically on table printing

- All "infos" -> refactor into classes (example: see HDD) and use them to query

- Rich util etc could be deleted then (in the util dir)

Imipenem committed:
Started redesigning SI

- BaseInfo (not quite happy with naming though): Hold Info basically on table printing

- All "infos" -> refactor into classes (example: see HDD) and use them to query

Imipenem commented on system_intelligence/query.py:
```diff
@@ -3,13 +3,14 @@
 from .cpu_info import query_cpu, print_cpu_info  # noqa F401
 from .gpu_info import query_gpus, print_gpus_info  # noqa F401
-from .hdd_info import print_hdd_info, query_hdd  # noqa F401
+from .hdd_info import HddInfo # noqa F401
```
We wont need the import here

Imipenem committed:
Restructured all "querier"

- need to restructure tests as well

- need to think of a neat solution since os, host are very small classes (maybe integrate somehow in base class)

Imipenem committed:
GpuInfo -> GpusInfo

Zethson commented on system_intelligence/cpu_info.py:
```diff
@@ -35,99 +31,101 @@
+            'clock': f'{str(clock_current)} MHz',
+            'clock_min': f'{str(clock_min)} MHz',
+            'clock_max': f'{str(clock_max)} MHz',
+            'cache': str(cache)}
```
In such cases I prefer the `}` on a new line

Zethson commented on system_intelligence/gpus_info.py:
```diff
@@ -0,0 +1,112 @@
+    5: 'Maxwell',
+    6: 'Pascal',
+    7: 'Volta',
+    8: 'Ampere'
```
We could and should include the latest GPU generation here.
It requires some googling to get the specs, but absolutely required

Imipenem replied:
Will do this!

Zethson replied:
Don't forget it :)

We should also ensure that the case of a user having a GPU that is not (yet) in this dictionary.
It should return a reasonable default aka "Unknown" or something like that

Imipenem committed:
First Draft MacOS CPU query (wont run)

Imipenem committed:
Fixed subprocess pipe for MacOS

Imipenem committed:
Added MacOS CPU info support

Imipenem committed:
RAM MacOS support

Imipenem commented on system_intelligence/ram_info.py:
```diff
@@ -1,190 +1,229 @@
 try:
     import pyudev
-
     pyudev.Context()
```
On MacOS actually, `pyudev` seems not available. It catches the `ImportError` but I get an Info along the lined `Exception ignore in some pyudev file AttributeError`. It does not cause the program to fail but its rather annoying. I tried to catch it as well but this did not work. I also tried a nested `try except` in the first `except` statement but this completly prohibits any printing ;(.

Any ideas how to fix this @Zethson ?

Imipenem commented on system_intelligence/ram_info.py:
```diff
@@ -1,190 +1,229 @@
+    def __init__(self):
+        super().__init__()
+        self.RAM_TOTAL = psutil is not None
+        self.OS = platform
```
Thinking of making `OS` a variable of `BaseInfo`. Since this information is not only needed here, but elsewhere (for example CPU)

Imipenem committed:
Fixed ignored exception, refactored OS into base class

- fixed the print of an ignored exception, if pyudev could not be imported

- the name of the users OS is now refactored into base class

- increased lint line length to 160

- some flak8 lint fixes

Imipenem committed:
Fixed output file path when generating html table

- output path parameter now allows path to define where the html file should be generated

- removed unused import statement from hdd,ram,cpu

- fixed some lint errors

Imipenem committed:
Added exclude option, fixed tests, flake8 lint, added descriptions

Imipenem committed:
Fixed tests, refactored util into base class

Imipenem committed:
Use decorator for registering query classes

- no need for importlib and import at runtime

Imipenem committed:
Removed decorator ;)

Imipenem committed:
Test macOS + added run to build WF

- run especially to check, whats broke on windows atm

Imipenem committed:
MacOS host now also queries model name

- removed windows for now from WFs (will add later if Windows is supported)

Imipenem committed:
Fix ubuntu host query

Imipenem committed:
Refactored software query code, swap mem now printed formatted

- SW querier no only calls pip once for packages installed to the current env, resulting in a nice speed boost overall

- the swap memory size is now formatted like CPU cache and RAM instead of bytes only

- renamed is_command_accessible variable and replaced click prints with rich print

Imipenem committed:
Software querier: refactored if statements and added comments

- added myself to Authors

Imipenem committed:
[First Draft] Windows Support

- only RAM query needed to be fixed

- html output tested (working)

- added windows to build and run workflow

Zethson commented on .github/workflows/build_package.yml:
```diff
@@ -21,3 +21,6 @@ jobs:
 
```
Let's add Python 3.9 to the matrix

Imipenem replied:
Instead of 3.7 or on top?

Zethson commented on AUTHORS.rst:
```diff
@@ -9,5 +9,6 @@ Development Lead
 ------------
+* Philipp Ehmele <philipp_ehm@protonmail.com>
 
 None yet. Why not be the first?
```
This can be removed if you add yourself here :)

Zethson commented on system_intelligence/base_info.py:
```diff
@@ -0,0 +1,93 @@
+
+    def print_table(self):
+        """
+        Print the result table
```
```suggestion
        Print the result table to stdout
```

Zethson commented on system_intelligence/cpu_info.py:
```diff
@@ -25,109 +22,106 @@
+
+        return {
+            'vendor_id_raw': cpu.get('vendor_id_raw'),
+            'hardware_raw': cpu.get('hardware_raw') if self.OS == 'linux' else '',
```
else maybe something like "NA"?

Imipenem replied:
Yeah

Zethson commented on system_intelligence/gpus_info.py:
```diff
@@ -0,0 +1,111 @@
+            return {
+                'architecture': compute_capability_to_architecture[compute_capability[0]],
+                'brand': device.name(),
+                'compute_capability': str(float('.'.join(str(_) for _ in compute_capability))),
```
That's interesting. It should usually only every be a single compute capability. Mixing different GPU generations works almost never. -> Even when using 2 GPUs the compute capability should only be a single number

Imipenem replied:
Yeah, I didn't change this code here (besides the different line number), but I also do not habe the possibility of testing if I remove this (atm)

Zethson replied:
I could test it

Imipenem replied:
@Zethson  could pls remove and test it quick?

Zethson replied:
Think I had a wrong thought. This should be all right.

Zethson commented on system_intelligence/hdd_info.py:
```diff
@@ -1,108 +1,102 @@
+            for device_ in itertools.chain([device], device.ancestors):
+                try:
+                    hdd['model'] = device_.attributes.asstring('model')
+                    break
```
what is this break for?

Imipenem replied:
Did not change the code here, so also have to take a look at the code

Zethson commented on system_intelligence/ram_info.py:
```diff
@@ -1,190 +1,261 @@
+        # split the BANKs (like "slots") into different items
+        ram_banks = [e.strip() for e in ram_info.split('BANK') if e][1:]
+
+        # for each RAM slot, extract details
```
```suggestion
        # for each RAM slot extract details
```

Zethson commented on system_intelligence/ram_info.py:
```diff
@@ -1,190 +1,261 @@
+
+        # for each RAM slot, extract details
+        for i in range(0, len(ram_banks)):
+            ram_slot_details = [e.strip() for e in ram_banks[i].split('\n') if e]
```
what is e?

Zethson commented on system_intelligence/ram_info.py:
```diff
@@ -1,190 +1,261 @@
+        # for each RAM slot, extract details
+        for i in range(0, len(ram_banks)):
+            ram_slot_details = [e.strip() for e in ram_banks[i].split('\n') if e]
+            ram_slot_details_set = {el.split(':')[0].strip(): el.split(':')[1].strip() for el in ram_slot_details}
```
what is el?

Zethson commented on system_intelligence/ram_info.py:
```diff
@@ -1,190 +1,261 @@
+        for i in range(0, len(ram_banks)):
+            ram_slot_details = [e.strip() for e in ram_banks[i].split('\n') if e]
+            ram_slot_details_set = {el.split(':')[0].strip(): el.split(':')[1].strip() for el in ram_slot_details}
+            ram['banks']['BANK ' + ram_slot_details[0]] = {k: v for k, v in ram_slot_details_set.items() if k in
```
still no a fan of k, v but we can keep it here :)

Imipenem replied:
Will rename, something along `atribute, value`

Zethson commented on system_intelligence/ram_info.py:
```diff
@@ -1,190 +1,261 @@
+            slot_attributes = slot.split(',')
+            bank_name = slot_attributes[1]
+            ram['banks'][bank_name] = {k: v for k, v in zip(['Capacity', 'Description', 'Manufacturer', 'Speed'],
+                                                         slot_attributes[2:])}
```
looks like weird formatting to me (with the line above)

Zethson reviewed (changes requested):
> Swap size: 

I have no swap, therefore it is empty. It should return "NA" or something.

Code breaks:

```
Traceback (most recent call last):
  File "/home/zeth/anaconda3/envs/systemintelligence/bin/system-intelligence", line 8, in <module>
    sys.exit(main())
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/system_intelligence/system_intelligence_cli.py", line 57, in main
    query_and_export(query_scope=scope,
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/system_intelligence/query.py", line 18, in query_and_export
    info = query(query_scope, verbose, **kwargs)
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/system_intelligence/query.py", line 43, in query
    query_info = getattr(instance, f'query_{query}')()
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/system_intelligence/cpu_info.py", line 54, in query_cpu
    cache = dict(self._get_cache_sizes(cpu))
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/system_intelligence/cpu_info.py", line 116, in _get_cache_sizes
    return {lvl: CpuInfo._get_cache_size(self, lvl, cpuinfo_data) for lvl in range(1, 4)}
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/system_intelligence/cpu_info.py", line 116, in <dictcomp>
    return {lvl: CpuInfo._get_cache_size(self, lvl, cpuinfo_data) for lvl in range(1, 4)}
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/system_intelligence/cpu_info.py", line 110, in _get_cache_size
    return CpuInfo.format_bytes(cache_size)
  File "/home/zeth/anaconda3/envs/systemintelligence/lib/python3.8/site-packages/system_intelligence/base_info.py", line 65, in format_bytes
    size = int(size)
ValueError: invalid literal for int() with base 10: '128 KiB'

```

Zethson commented:
Add 3.9 on top, please. The CI is lightweight enough  Sunday, 03 January 2021, 05:09PM +01:00 from Philipp Ehmele  notifications@github.com :

>@Imipenem commented on this pull request.
>----------------------------------------------------------------------
>In  .github/workflows/build_package.yml :
> @@ -21,3 +21,6 @@ jobs:
> 
>
>Instead of 3.7 or on top?
>—
>You are receiving this because you were mentioned.
>Reply to this email directly,  view it on GitHub , or  unsubscribe .

Imipenem committed:
PR review, base10 vs base2 conversion for diff OS, 3.9py for build WF

- memory sizes (for Ubuntu except Disk size) for Ubuntu and MacOS are now computed in base10

- added correct units for memory sizes in base2 units

- adressed PR feedback

- added python 3.9 to build WF

Imipenem committed:
Fixed WFs

- plistlib API was removed and updated with python3.9

- fixed tests and linting

Imipenem committed:
Fixed plistlib modelname query macOS for python 3.9

Imipenem committed:
Bump version from 1.2.4 to 2.0.0

Imipenem committed:
Updated Changelog

Imipenem committed:
Added missing column for GPU

Imipenem committed:
Removed double print

Imipenem committed:
Linting
